app/controllers/public/divisions_controller.rb

```diff
@@ -0,0 +1,30 @@
+class Public::DivisionsController < Public::PublicController
+  layout "public/main"
+
+  rescue_from Pundit::NotAuthorizedError, with: :user_not_authorized
+
+  def show
+    @params = {status: params[:status], pg: params[:pg]}
+    @division = Division.find_by(short_name: params[:short_name])
+    @status_filter = [:active]
+    if @params[:status] == "all"
+      @status_filter = [:active, :completed]
+    elsif @params[:status]
+      @status_filter = [@params[:status]]
+    end
+    @selected_division = @division
+    authorize @division
+    # loan scope handles filtering loans based on whether their division is public,
+    # only completed or active loans, and their public level value
```

Comment:
Thanks for this comment. I'm not sure if you want to get into this but I'll offer it. I think the responsibility for determining which divisions' loans to include does lie with the Policy class. I think the way to implement it might look something like:

```
LoanPolicy::Scope.new(nil, Loan).publicly_visible_for_division(@division).where(status_value: ...)...
```

You'd get rid of the implicit call to the `publicly_visible` method and call this new one directly, which would contain the old logic plus the restriction to self and descendants.

The trick here is knowing that you can define custom methods on the Scope object and just bypass the `policy_scope` shortcut, which only knows how to call `resolve` with no arguments.

Again, I think it would be fine to leave it here if you really want to move on but if you have space I'd suggest refactoring and adjusting the LoanPolicy spec.